### ISSUE TITLE
long formula convert to decimal expression

### ISSUE THREAD

**Issue by huahuayu** [shopspring/decimal#268]
hi, I wonder if there's any way to convert a long formula into decimal expression, for example

```
-(A * D - B * C * E + C * D * E + C * E * (-math.sqrt(D * (A ** 2 * D * E ** 2 - 2 * A ** 2 * D * E + A ** 2 * D + 4 * A * B * C * E ** 2 + 2 * A * C * D * E ** 3 - 2 * A * C * D * E ** 2 + C ** 2 * D * E ** 4)) / (2 * C * E ** 2) - (A * D * E - A * D - 2 * B * C * E ** 2 + C * D * E ** 2) / (2 * C * E ** 2))) / D - 1
```

I tried it for several times by hand writing, there's no readability and easy to get it into mass. If only there's a convert tool.

**Comment by mwoss** [comment]
HI @huahuayu. Do you mean converting math formula (string representation) to code?
For example: `A * B + C` to `A.Mul(B).Add(C)`? If so this library does not support such a feature, but I believe it would be quite fun side project that could use `decimal` as dependency :)

**Comment by huahuayu** [comment]
> Do you mean converting math formula (string representation) to code?
> For example: A * B + C to A.Mul(B).Add(C)?

Exactly! I am look for helper tools like your description. For long formula it's so painful to convert it to `decimal`.

**Comment by mwoss** [comment]
I don't know any of such tools, unfortunately, but I believe there are a few of them already somewhere on the internet in different languages (at least for simple math formulas). Try to find them and rewrite them in Go using `decimal` :D
